### PR TITLE
225 error 500 camel case

### PR DESCRIPTION
**Pull request by Nightmare75**
In folder /www/modules/oyst/controllers/front, all classes name were with Upercase letter.
I replaced OystPayment**R**eturnModuleFrontController by OystPayment**r**eturnModuleFrontController

@quentinjunior 
[oyst.zip](https://github.com/oystparis/oyst-prestashop/files/1125097/oyst.zip)

### CODE REVIEW

**Comment by vschoener** [comment]
That's not a fixe... PHP is not camel case sensitive at all ! The real problem is due to the getModuleLink call with bad string request.

Ex: this->context->link->getModuleLink('oyst', 'paymentNotification'), it should be this->context->link->getModuleLink('oyst', 'paymentnotification').

**Comment by quentinjunior** [comment]
détail erreur sur phimedical:
AH01068: Got bogus version 32, referer: https://www.phimedical.fr/module/oyst/paymentReturn?id_cart=13003&key=3f6a21193ac1783bb766f2c0ab48021f
 
(22)Invalid argument: AH01075: Error dispatching request to : (passing brigade to output filters), referer: https://www.phimedical.fr/module/oyst/paymentReturn?id_cart=13003&key=3f6a21193ac1783bb766f2c0ab48021f
 
Et un warning:
 
AH01071: Got error 'PHP message: PHP Warning: include_once(/var/www/vhosts/phimedical.fr/httpdocs/modules/oyst/controllers/front/paymentNotification.php): failed to open stream: No such file or directory in /var/www/vhosts/phimedical.fr/httpdocs/override/classes/Dispatcher.php on line 227\nPHP message: PHP Warning: include_once(): Failed opening '/var/www/vhosts/phimedical.fr/httpdocs/modules/oyst/controllers/front/paymentNotification.php' for inclusion (include_path='/var/www/vhosts/phimedical.fr/httpdocs/tools/htmlpurifier/standalone:.:/opt/plesk/php/5.6/share/pear') in /var/www/vhosts/phimedical.fr/httpdocs/override/classes/Dispatcher.php on line 227\nPHP message: PHP Fatal error: Class 'oystpaymentNotificationModuleFrontController' not found in /var/www/vhosts/phimedical.fr/httpdocs/classes/controller/Controller.php on line 134\n'

**Comment by vschoener** [comment]
@quentinjunior yeah, as we talked about, the problem remains the same. The url should be 
`https://www.phimedical.fr/module/oyst/paymentreturn?id_cart=13003&key=3f6a21193ac1783bb766f2c0ab48021f `

PrestaShop will take the last slug `/paymentReturn`and will try to load this file dynamically looking inside the oyst module folder and specifically inside the controller/front folder.

For this case : `paymentReturn` will be `paymentReturn.php` and then the generated path  `'..'/modules/oyst/controller/front/paymentReturn.php` 

But the slug has to be `paymentreturn` that will become `paymentreturn.php` and then the path : `'..'/modules/oyst/controller/front/paymentreturn.php`

**Comment by Nightmare75** [comment]
@quentinjunior  @vschoener 

The previous .zip version was testing on a Prestashop 1.6 and no error was founded.
I also check all the urls before commit : 

array (size=4)
  'notification' => string 'http://9c278c6c.ngrok.io/prestashop_1.6.1.4/index.php?fc=module&module=oyst&controller=paymentnotification&key=70160baef920730dc881b1101d040ff1-20170703063218&ch=a060c40c7c6b291d85bf78b5f087faef' (length=194)
  'cancel' => string 'http://9c278c6c.ngrok.io/prestashop_1.6.1.4/index.php?fc=module&module=oyst&controller=paymenterror' (length=99)
  'error' => string 'http://9c278c6c.ngrok.io/prestashop_1.6.1.4/index.php?fc=module&module=oyst&controller=paymenterror' (length=99)
  'return' => string 'http://9c278c6c.ngrok.io/prestashop_1.6.1.4/index.php?fc=module&module=oyst&controller=paymentreturn&id_cart=12&key=f441018cd7b249d0faf7d9f5165b17aa' (length=148)

Here is the last version : with lowercase for classes : 
- $this->context->link->getModuleLink('oyst', 'paymentnotification')
- $this->context->link->getModuleLink('oyst', 'paymentreturn')

[oyst.zip](https://github.com/oystparis/oyst-prestashop/files/1127554/oyst.zip)

**Comment by Nightmare75** [comment]
Test successfull with previous version and latest version : 

image

Back Office : 

image

**Comment by Nightmare75** [comment]
Test of Oyst Button successful on the previous and latest version : http://www.madhuri-nasr.fr/en/abaya/1327-abaya-mira.html

File: views/js/OystOneClick.js [hunk 1]
```diff
@@ -62,7 +61,7 @@ function OystOneClick(url, productId) {
             available = false;
         }
 
-        if (productExported.isExported && available) 
+        if (available)
```
Comment (vschoener) [inline views/js/OystOneClick.js]: That's not the way to know if the product is available or not, using the css that way is bad.

Comment (vschoener) [inline views/js/OystOneClick.js]: For PHP and Js, try to follow this guideline, http://www.php-fig.org/psr/psr-2/

Comment (vschoener) [inline views/js/OystOneClick.js]: Why removing the condition with isExported ? This one is set on the server side to know if the product has been sent to oyst..

**Pull request closed without merging**